lib/vector-core/src/schema/definition.rs

```diff
@@ -1,9 +1,326 @@
-#[derive(Debug, Clone, PartialEq)]
-pub struct Definition;
+use std::collections::{HashMap, HashSet};
+
+use lookup::LookupBuf;
+use value::{
+    kind::{merge, nest, Collection, Field, Unknown},
+    Kind,
+};
+
+/// The definition of a schema.
+///
+/// This struct contains all the information needed to inspect the schema of an event emitted by
+/// a source/transform.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Definition {
+    /// The collection of fields and their types stored in the event.
+    collection: Collection<Field>,
+
+    /// Semantic meaning assigned to fields within the collection.
+    ///
+    /// The value within this map points to a path inside the `collection`. It is an invalid state
+    /// for there to be a meaning pointing to a non-existing path in the collection.
+    meaning: HashMap<String, LookupBuf>,
```

Comment:
Note that semantic meanings are currently untyped (strings), but I'll be migrating this to an enum type that includes a set of known/pre-defined meanings (e.g. `Message`, `Hostname`, etc), and a `Custom(String)` variant.

This will happen in a follow-up PR as schema support is added to the Datadog agent source.